### PR TITLE
Don't let Mux.Route(...) stomp on already handled routes

### LINKED ISSUE #336: Cannot combine handlers with a matching Route
```go
r := NewRouter()

// A handler that does not use middleware
r.Post("/parent", postHandler)

r.Group(func(r Router) {
	r.Use(mw)

	r.Route("/parent", func(r Router) {
		r.Get("/", getHandler)
		// ... other routes at or under `/parent` that use the middleware
	})
})
```

Requests to `GET /parent` hit the middleware and handler, as expected.

Requests to `POST /parent` hit the middleware, and not the handler (`405 Method Not Allowed` is returned). Neither of those things are expected.

Simplifying to this produces the same result (`GET` handler works, `POST` handler does not):

```go
r := NewRouter()
r.Post("/parent", postHandler)

r.Route("/parent", func(r Router) {
	r.Get("/", getHandler)
	// ...
})
```

Here is a test demonstrating the issue:

```go

func TestRouteAndStandalone(t *testing.T) {
	handlerCalled := false
	handler := func(w http.ResponseWriter, r *http.Request) {
		// t.Logf("Called %s handler", r.Method)
		handlerCalled = true
	}

	r := NewRouter()
	r.Post("/parent", handler)

	r.Route("/parent", func(r Router) {
		r.Get("/", handler)
	})

	ts := httptest.NewServer(r)
	defer ts.Close()

	resp, _ := testRequest(t, ts, http.MethodPost, "/parent", nil)
	if resp.StatusCode != http.StatusOK {
		t.Errorf("Failed to reach POST handler: %s", resp.Status)
	}

	if !handlerCalled {
		t.Error("Expected to reach POST handler, but did not")
	}

	handlerCalled = false

	resp, _ = testRequest(t, ts, http.MethodGet, "/parent", nil)
	if resp.StatusCode != http.StatusOK {
		t.Errorf("Failed to reach GET handler: %s", resp.Status)
	}

	if !handlerCalled {
		t.Error("Expected to reach GET handler, but did not")
	}
}
```

The output is:

```text
$ go test -v -run TestRouteAndStandalone
=== RUN   TestRouteAndStandalone
--- FAIL: TestRouteAndStandalone (0.00s)
	mux_test.go:1311: Failed to reach POST handler: 405 Method Not Allowed
	mux_test.go:1315: Expected to reach POST handler, but did not
FAIL
exit status 1
FAIL	github.com/go-chi/chi	0.015s
```

This is true of v3.3.2, and master @ 0c5e7ab.

### PR DESCRIPTION
**Pull request by dcormier**
Fixes #336

### CODE REVIEW

**Comment by mcastilho** [comment]
This is good...

File: tree.go [hunk 1]
```diff
@@ -345,7 +345,13 @@ func (n *node) setEndpoint(method methodTyp, handler http.Handler, pattern strin
 		h.pattern = pattern
 		h.paramKeys = paramKeys
 		for _, m := range methodMap {
-			h := n.endpoints.Value(m)
+			h, ok := n.endpoints[m]
+			if ok {
+				// Don't stomp on already mapped endpoints
+				continue
```
Comment (thedevsaddam) [inline tree.go]: It should panic, instead of silently skipping

Comment (thedevsaddam) [inline tree.go]: Isn't is better to panic, instead of silently skipping?

Comment (thedevsaddam) [inline tree.go]: Isn't it better to panic, instead of silently skipping?

**Comment by pkieltyka** [comment]
thanks for the PR, however, the intended design of chi is to allow overriding routes

**Pull request closed without merging**